sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/impl/DiscountServiceImpl.java

```diff
@@ -62,4 +71,30 @@ public DiscountDto update(DiscountDto discount) {
     public void delete(Long id) {
         discountDAO.deleteById(id);
     }
+
+    @Override
+    public Page<DiscountDto> getByCriteria(DiscountSearchCriteria searchCriteria) {
+        List<Discount> result = new ArrayList<>();
+        long rate;
+        PageRequest pageRequest = searchCriteria.getPageRequest();
+        Set<Tag> tags = tagDAO.getTagsByNameIn(searchCriteria.getTags());
+        String searchText = searchCriteria.getSearchText();
+        String sortBy = searchCriteria.getSortBy();
+        List<Discount> discounts = discountDAO.getByNameContaining(searchText);
+        discounts = discounts.stream().filter(x -> x.getTags().containsAll(tags)).collect(Collectors.toList());
+        if (sortBy.equals("Top rated")) {
```

Comment:
We should use one SQL call to database instead of several

Reply:
it's call to two different DAO because i don't know how to do it by using only DiscountDAO

Reply:
and then a coincidence check